### PR TITLE
2.0-preview

### PR DESCRIPTION
**Pull request by e111077**
DO NOT TAG

### CODE REVIEW

**Comment by AlexNasonov** [comment]
https://github.com/firebase/polymerfire/issues/170 - check this issue - it's still a bug in 2.0-preview

**Comment by e111077** [comment]
Heya, Alex. I'm trying to give the proper credit to the author by trying to merge his PR, but I can't merge his branch due to some conflicts. Awaiting his response.

**Comment by bmwalters** [comment]
@e111077 You could do the following:
```sh
git remote add wintersandroid git@github.com:wintersandroid/polymerfire.git
git fetch --all
git merge --squash wintersandroid/2.0-preview
# fix merge conflict in bower.json
git add bower.json
git commit
```

File: bower-1.x.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (notwaldorf) [inline bower-1.x.json]: This should be in .gitignore

File: bower.json [hunk 2]
```diff
@@ -18,12 +18,31 @@
...
+    "1.x": {
+      "dependencies": {
+        "polymer": "Polymer/polymer#^1.2",
+        "firebase": ">= 3.5.1 < 4.0",
+        "app-storage": "polymerelements/app-storage#^0.9.0"
+      },
+      "devDependencies": {
+        "iron-component-page": "PolymerElements/iron-component-page#^1.0",
+        "note-app-elements": "polymerlabs/note-app-elements",
+        "test-fixture": "polymerelements/test-fixture#^1.0.0",
```
Comment (notwaldorf) [inline bower.json]: don't think we need `test-fixture` anymore (here and above)

File: demo/firebase-messaging.html [hunk 3]
```diff
@@ -60,6 +60,9 @@
       window.addEventListener('WebComponentsReady', function() {
         if (navigator.serviceWorker) {
           navigator.serviceWorker.register('firebase-messaging-sw.js').then(function(sw) {
+            var app = document.getElementById('app');
+            // polymer 2.0 compatibility
+            app = app.$ ? app : app.parentNode;
```
Comment (notwaldorf) [inline demo/firebase-messaging.html]: Huh, how? Isn't `$` in both things? Maybe you need `Polymer.Element` somewhere instead

File: demo/firebase-messaging.html [hunk 4]
```diff
@@ -70,41 +73,50 @@
...
         return JSON.stringify(val || null, null, 2);
       }
 
       app.logit = function(e) {
-        app.$.log.innerHTML = JSON.stringify(e.detail.message) + "\n" + app.$.log.innerHTML;
+        this.$.log.innerHTML = JSON.stringify(e.detail.message) + "\n" + app.$.log.innerHTML;
       }
+
+      // polymer 2.0 compatibility
+      var domBind = app.parentNode;
```
Comment (notwaldorf) [inline demo/firebase-messaging.html]: Should we wrap this in a `if (Polymer.Element)` check?

File: firebase-app.html [hunk 5]
```diff
@@ -8,6 +8,12 @@
 
 <link rel="import" href="../polymer/polymer.html">
 <link rel="import" href="firebase-app-script.html">
+<!-- TODO(cdata): Remove all of the imports below this line when it is possible
+     to lazy load Firebase scripts alongside Custom Elements v1 upgrade order -->
+<link rel="import" href="firebase-database-script.html">
```
Comment (notwaldorf) [inline firebase-app.html]: What's the status of this TODO? @cdata

File: test/firebase-query.html [hunk 6]
```diff
@@ -353,7 +359,8 @@
             return setFirebaseValue(
                 query.path + '/' + query.data[0].$key + '/val', object.val + 1);
           }).then(function() {
-            expect(domRepeat._instances[0].item).to.be.eql(query.data[0]);
+            var instances = domRepeat._instances || domRepeat.__instances;
```
Comment (notwaldorf) [inline test/firebase-query.html]: uhhh i think `domRepeat._instances || domRepeat.__instances` is always `domRepeat._instances` so you don't have to or it? :)

File: bower-1.x.json (hunk 1 above)
Comment (e111077) [inline bower-1.x.json]: yuppp

File: bower.json (hunk 2 above)
Comment (e111077) [inline bower.json]: noppppp

File: demo/firebase-messaging.html (hunk 3 above)
Comment (e111077) [inline demo/firebase-messaging.html]: well in Polymer 1 `<dom-bind>` wont have `.$` and in Polymer 2 `<template is="dom-bind"` won't have `.$`

Comment (e111077) [inline demo/firebase-messaging.html]: Will do Polymer.Element instead

File: test/firebase-query.html (hunk 6 above)
Comment (e111077) [inline test/firebase-query.html]: I just rechecked `domRepeat._instances` is undefined in Polymer 2 and this propertly seems to set `instance` to the value of `domRepeat.__instances`

Comment (e111077) [inline test/firebase-query.html]: Polymer 1 seems to do the right thing as well

**Comment by notwaldorf** [comment]
@e111077 are you pinky swearing these tests are green? i am spoiled and don't see a travis being run :sob:

**Comment by notwaldorf** [comment]
Pretty sure this is 🎉 🍰 🆗

File: bower.json [hunk 7]
```diff
@@ -18,12 +18,29 @@
     "/test/"
   ],
   "dependencies": {
-    "polymer": "Polymer/polymer#^1.2",
+    "polymer": "polymer/polymer#2.0.0-rc.3",
```
Comment (notwaldorf) [inline bower.json]: This should be a ^

**Pull request merged**